software/nanoplot/main.nf

```diff
@@ -0,0 +1,46 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process NANOPLOT {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::nanoplot=1.32.1" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/nanoplot:1.32.1--py_0"
+    } else {
+        container "quay.io/biocontainers/nanoplot:1.32.1--py_0"
+    }
+
+    input:
+    tuple val(meta), path(fastq), path(summary_txt)
```

Comment:
Developers will almost certainly want to run this module on different combinations of inputs and it would be good if can have the flexibility to run on one or both of `fastq` and `summary_txt`. You could change that to:

```
inpur:
tuple val(meta), path(ontfile)
```

We could have something that checks the file extension and adds the appropriate flag? 
https://github.com/nf-core/modules/blob/13bd67ed8aca271a260c78ca41afe9e1f5c217d7/software/vcftools/main.nf#L115-L117

Reply:
Be great to have separate tests for both though.

Reply:
Thanks for the suggestion! I am now using `ontfile` for accepting either a `fastq` or a `summary.txt` input. There are two separate tests for this.